### PR TITLE
Updating Errors and Auth

### PR DESCRIPTION
**Pull request by jameszhang244**
Updating Errors and Auth

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,11 +10,15 @@ keywords = ["Palantir", "Foundry", "SDK", "Client", "API"]
 packages = [{ include = "foundry" }]
 
 [tool.poetry.dependencies]
+annotated-types = ">=0.7.0"
+expects = ">=0.9.0"
+mockito = ">=1.5.1"
+pydantic = "^2.1.0"
+pytest = ">=7.4.0"
+pytest-asyncio = ">=0.23.0"
```
Comment (afloren-palantir) [inline pyproject.toml]: test dependencies should be in their own group https://python-poetry.org/docs/managing-dependencies/ i.e., they aren't required to install and use the package

**Review by afloren-palantir — DISMISSED** [review]

File: pyproject.toml (hunk 1 above)
Comment (afloren-palantir) [reply pyproject.toml]: will also need to update CI logic to install test deps

**Review by afloren-palantir — DISMISSED** [review]

**Comment by afloren-palantir** [comment]
woo looks like I was finally added to the org

**Comment by jameszhang244** [comment]
omg 😭 I didn't know it would dismiss the review

**Comment by afloren-palantir** [comment]
👍

**Pull request merged**